From 5ad9c1422c1322adaeb2da3abe88305d5315267b Mon Sep 17 00:00:00 2001
From: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
Date: Wed, 4 Feb 2026 12:59:07 +0700
Subject: [PATCH 18/22] vspm_main: Force runtime suspend/resume when system 
 suspend/resume

Whenever the system suspend/resume, we will call the power on/off
sequence corresponding to PM runtime suspend/resume

Signed-off-by: Huynh Thanh Hung <hung.huynh.wz@renesas.com>
---
 vspm-module/files/vspm/drv/vspm_main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/vspm-module/files/vspm/drv/vspm_main.c b/vspm-module/files/vspm/drv/vspm_main.c
index 418b41f..c72da33 100644
--- a/vspm-module/files/vspm/drv/vspm_main.c
+++ b/vspm-module/files/vspm/drv/vspm_main.c
@@ -484,6 +484,7 @@ static int vspm_pm_suspend(struct device *dev)
 			(void)vspm_suspend(pdrv);
 		}
 	}
+	pm_runtime_force_suspend(dev);
 
 	return 0;
 }
@@ -492,6 +493,7 @@ static int vspm_pm_resume(struct device *dev)
 {
 	struct vspm_drvdata *pdrv = p_vspm_drvdata;
 
+	pm_runtime_force_resume(dev);
 	if (atomic_sub_return(1, &pdrv->suspend) == 0) {
 		/* last resume */
 		if (atomic_read(&pdrv->counter) != 0) {
-- 
2.25.1

